google-cloud-contrib/google-cloud-logging-logback/src/main/java/com/google/cloud/logging/contrib/LoggingAppender.java

```diff
@@ -0,0 +1,266 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.logging.contrib;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.UnsynchronizedAppenderBase;
+import com.google.cloud.MonitoredResource;
+import com.google.cloud.logging.LogEntry;
+import com.google.cloud.logging.Logging;
+import com.google.cloud.logging.Logging.WriteOption;
+import com.google.cloud.logging.LoggingEnhancer;
+import com.google.cloud.logging.LoggingOptions;
+import com.google.cloud.logging.MonitoredResourceUtil;
+import com.google.cloud.logging.Payload;
+import com.google.cloud.logging.Severity;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * <a href="https://logback.qos.ch/">Logback</a> appender for StackDriver Cloud Logging.
+ *
+ * <p>Appender configuration in logback.xml:
+ * <ul>
+ * <li>&lt;appender name="CLOUD" class="com.google.cloud.logging.contrib.LoggingAppender"&gt;</li>
+ * <li>&lt;log&gt;application.log&lt;/log&gt; (Optional, defaults to "java.log" : Stackdriver log name)</li>
+ * <li>&lt;level&gt;ERROR&lt;/level&gt; (Optional, defaults to "INFO" : logs at or above this level)</li>
+ * <li>&lt;flushLevel&gt;WARNING&lt;/flushLevel&gt; (Optional, defaults to "ERROR")</li>
+ * <li>&lt;resourceType&gt;&lt;/resourceType&gt; (Optional, auto detects on App Engine Flex, Standard, GCE and
+ * GKE, defaults to "global".
+ * See <a href="https://cloud.google.com/logging/docs/api/v2/resource-list">supported resource
+ * types</a></li>
+ * <li>&lt;enhancers&gt;com.example.enhancer1, com.example.enhancer2&lt;/enhancers&gt;</li>
+ * <li>(Optional) add custom labels to log entries using {@link LoggingEnhancer} classes.</li>
+ * <li>&lt;/appender&gt;</li>
+ * </ul>
+ */
+public class LoggingAppender extends UnsynchronizedAppenderBase<ILoggingEvent> {
+
+  private static final String LEVEL_NAME_KEY = "levelName";
+  private static final String LEVEL_VALUE_KEY = "levelValue";
+
+  private volatile Logging logging;
+  private List<LoggingEnhancer> loggingEnhancers;
+  private WriteOption[] defaultWriteOptions;
+
+  private Level flushLevel;
+  private String log;
+  private Level logLevel;
+  private String resourceType;
+  private String enhancers;
+
+  /**
+   * Batched logging requests get immediately flushed for logs at or above level.
+   *
+   * <p>Defaults to Error if not set.
+   *
+   * @param flushLevel Logback log level
+   */
+  public void setFlushLevel(Level flushLevel) {
+    this.flushLevel = flushLevel;
+  }
+
+  /**
+   * Sets the log filename, defaults to java.log.
+   *
+   * @param log filename
+   */
+  public void setLog(String log) {
+    this.log = log;
+  }
+
+  /**
+   * Sets the minimum logging level, defaults to INFO
```

Comment:
It looks like the Logback way of doing this is through [filters](https://logback.qos.ch/manual/filters.html).